src/crimson/net/Socket.h

```diff
@@ -39,6 +44,12 @@ class Socket
       // performance. see seastar::net::connected_socket::output()
       out(socket.output(65536)) {}
 
+  ~Socket() {
+#ifndef NDEBUG
```

Comment:
i don't think we need to guard `assert()` with `NDEBUG`. see http://man7.org/linux/man-pages/man3/assert.3.html